distributor/src/distributor/messenger.bal

```diff
@@ -0,0 +1,130 @@
+    string message  = "Results will be releasing soon for " + electionCode +  "/" + division + "(" + resultCode + ")";
+
+    foreach string targetMobile in mobileSubscribers {
+        if (targetMobile == INVALID_NO) {
+            continue;
+        }
+        var response = twilioClient->sendSms(sourceMobile, targetMobile, message);
+        if response is error {
+            log:printError(electionCode +  "/" + division + " message sending failed \'" + targetMobile +
+                           "\' due to error:" + <string> response.detail()?.message);
+        }
+    }
+}
+
+# Get the respective division name for a given resultCode.
+#
+# + resultCode - The predefined code for a released result
+# + return - The division name if resultCode is valid, otherwise nil
+function getDivision(string resultCode) returns string? {
+    return divisionCodeMap.hasKey(resultCode) ? divisionCodeMap.get(resultCode) : ();
+}
+
+# Validate and sanitize local mobile number into the proper format.(+94771234567).
+#
+# + mobileNo - User provided mobile number
+# + return - Formatted mobile number or the error
+function validate(string mobileNo) returns string|error {
+    string mobile = <@untained> mobileNo.trim();
+
+    boolean number = stringutils:matches(mobile, "^[0-9]*$");
+
+    if !number {
+        error err = error(ERROR_REASON, message = "Invalid mobile number. Given mobile number contains non numeric " +
+                                                  "characters: " + mobile);
+        return err;
+    }
+
+    if (mobile.startsWith("0") && mobile.length() == 10) {
+        return "+94" + mobile.substring(1);
+    }
+    if (mobile.startsWith("94") && mobile.length() == 11) {
+        return "+" + mobile;
+    }
+    // Allow only the local mobile numbers to register via public API. International number are avoided.
+    error err = error(ERROR_REASON, message = "Invalid mobile number. Resend the request as follows: If the " +
+                                    "mobile no is 0771234567, send request as \"/sms/94771234567\". ");
+    return err;
+}
+
+# Register recipient in the mobileSubscribers list and persist in the smsRecipients db table.
+#
+# + mobileNo - The recipient number
+# + return - The status of registration or operation error
+function registerAsSMSRecipient(string mobileNo) returns string|error {
+
+    foreach string recipient in mobileSubscribers {
+        if (recipient == mobileNo) {
+            error err = error(ERROR_REASON, message = "Registration failed: " + mobileNo + " is already registered");
+            return err;
+        }
+    }
+
+    // Persist recipient number in database
+    var status = dbClient->update(INSERT_RECIPIENT, mobileNo);
+    if status is error {
+        log:printError("Failed to persist recipient no in database: " + status.toString());
+        return status;
+    }
+    // Update the mobileSubscribers array
+    mobileSubscribers.push(mobileNo);
```

Comment:
Is this the function you introduced a lock for initially? There's the possibility of duplicates without right?

Reply:
Yes, also can lead to concurrent modifications. [Left a comment](https://github.com/ECLK/Results-Dist/pull/81/files#diff-23a02492b8512ee4547b798c5c549d3dR201) too. As per the Sanjiva's comment, we are registering/deregistering the numbers offline, so assume that we do not expect huge load. 

However I prefer having lock to the common resource to avoid every possible failure. But still the lock is --experimental
WDYT?